### PR TITLE
chore(dashboard): bump versions: Node.js to v20, Next.js to 14, typescript to 5.3.3

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As title. Hope this can make `Linting and checking validity of types` phase faster.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [x] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: dashboard/next.config.js [hunk 1]
```diff
@@ -19,6 +19,7 @@
  * @type {import('next').NextConfig}
  */
 const nextConfig = {
+  output: 'export',
```
Comment (BugenZhao) [inline dashboard/next.config.js:22]: cc @xxchan I remembered that there used to some discussion between us on this.

Comment (fuyufjh) [reply dashboard/next.config.js:22]: This is required to upgrading to next.js v14
https://nextjs.org/docs/pages/building-your-application/upgrading/version-14#v14-summary
https://nextjs.org/docs/app/building-your-application/deploying/static-exports

Comment (xxchan) [reply dashboard/next.config.js:22]: I remember `output: 'export'` will break `build-static`, and you suggested a solution `phase`

https://github.com/risingwavelabs/risingwave/pull/12421#discussion_r1330028759

Comment (xxchan) [reply dashboard/next.config.js:22]: Oh, it will break debug build, because it cannot rewrite to mock api conditionally.

Comment (xxchan) [reply dashboard/next.config.js:22]: Oh, you already changed it in https://github.com/risingwavelabs/risingwave/issues/14361 

I don't know whether there are any other issues now.

**Review by xxchan — APPROVED** [review]
rubber stamp

**Pull request merged**